### PR TITLE
[receiver/kafkareceiver] Support configuration of initial offset strategy

### PR DESCRIPTION
**Pull request by ueisele**
**Description:** Added the ability to configure the initial offset strategy to allow consuming form latest or earliest offset.

**Link to tracking Issue:** #14976

**Testing:** Unit tests to check configuration, e.g. that "latest" is still the default, and mapping to Sarama offset.

**Documentation:** Added documentation of new "initial_offset" configuration attribute.

### CODE REVIEW

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by MovieStoreGuy** [comment]
Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14976

File: receiver/kafkareceiver/config.go [hunk 1]
```diff
@@ -70,6 +73,11 @@ type Config struct {
 	MessageMarking MessageMarking `mapstructure:"message_marking"`
 }
 
+const (
+	OffsetLatest   string = "latest"
```
Comment (codeboten) [inline receiver/kafkareceiver/config.go:77]: are these needed outside of this package? If not do they need  to be public?

Comment (ueisele) [reply receiver/kafkareceiver/config.go:77]: @codeboten you are right. Its just needed within the package. I change the scope to private.

**Comment by TylerHelmuth** [comment]
@ueisele please rebase/merge in main to fix the failing loadtests.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**